extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -189,21 +189,25 @@ public final class HibernateOrmProcessor {
     void registerHibernateOrmMetadataForCoreDialects(
             BuildProducer<DatabaseKindDialectBuildItem> producer) {
         producer.produce(new DatabaseKindDialectBuildItem(DatabaseKind.DB2,
-                "org.hibernate.dialect.DB297Dialect"));
+                "org.hibernate.dialect.DB2Dialect"));
         producer.produce(new DatabaseKindDialectBuildItem(DatabaseKind.DERBY,
-                "org.hibernate.dialect.DerbyTenSevenDialect"));
+                "org.hibernate.dialect.DerbyDialect"));
```

Comment:
LGTM, the default (min) version is now 10.14.2:

https://github.com/hibernate/hibernate-orm/blob/f3833e71a973b8ae576b0b53a6bef7e9e16fd993/hibernate-core/src/main/java/org/hibernate/dialect/DerbyDialect.java#L114